go/extra/ba/README.md

```diff
@@ -0,0 +1,61 @@
+# Benchmark analysis
+
+This tool compares benchmark results generated by `oasis-test-runner` and
+corresponding `oasis-node` workers.
+
+## Usage
+
+Currently, `cmp` operation is implemented which compares the benchmark results
+of the last (`source`) and pre-last (`target`) test batches (also called
+*instances*):
+```bash
+$ ba/ba cmp \
+  --metrics.address http://localhost:9090
+```
+
+By default results of all `oasis-test-runner` tests and metrics will be
+compared. If you want to compare specific metric(s) and specific test(s),
+provide `-m` and `-t` flags respectively:
+```bash
+$ ba/ba cmp \
+  -m time \
+  -t multiple-runtimes \
+  --metrics.address http://localhost:9090
+```
+
+Comparison is done by considering *thresholds* for each metric. Currently,
+we support `avg_ratio` and `max_ratio` which correspond to the average and
+maximum ratio of metric values of all test runs in the benchmark batches. For
+each ratio, you can set the `max_threshold` and `min_threshold`. The former
+requires that the ratio should not be exceeded and the latter that the provided
+threshold must be reached. If not, `ba` exits with error code. Thresholds can
+be using the flag `--{min|max}_threshold.<metric name>.{avg|max}_ratio`.
+
+For example:
+```bash
+$ ba/ba cmp \
+  -m time \
+  -t multiple-runtimes \
+  --max_threshold.time.avg_ratio 1.1 \
+  --metrics.address http://localhost:9090
+```
+
+will require that the average duration of the test in the last benchmark batch
+should be at most 10\% slower than from the pre-last benchmark batch.
+
+If you are developing or improving a feature in a separate git branch, you will
+want to perform benchmarks and compare the results of your branch to the ones
```

Comment:
I think that even more helpful than the `ba` tool (or in addition to it), would be a grafana dashboard template that would display all relevant results from my run in a dashboard.

I think one way to achieve this would be: 
- update the `prometheus-dev.py` to also start a `grafana` container already connected with the existing `prometheus` instance
- the `grafana` should already contain a pre-defined dashboard that would display all relevant metrics

This would make it really easy to visually see what changed between runs and explore further the anomalies between runs.

Reply:
Yeah, this should be a separate issue. The idea of `ba` tool is actually to reduce the bash and python scripts needed to process benchmark results. `ba` also knows the internals of `oasis-test-runner` and `oasis-node` (expected test parameters, test names etc.). Can you open a separate issue for this? Thanks.